### ISSUE TITLE
As a backend director, I want to standardize our test directory format.

### ISSUE THREAD

**Issue by lucasmbrown-usds** [usds/justice40-tool#1076]
**Description**
For the tests on the scoring code, these live in `data/data-pipeline/data_pipeline/etl/score/tests`, next to the code that they're testing. 

However, for the national risk index ETL tests, these live in `data/data-pipeline/data_pipeline/tests/sources/national_risk_index` in a directory separated from the code. 

We should choose one approach and standardize it by fixing the non-conforming test directory/directories.

**Comment by esfoobar-usds** [comment]
It is more or less an accepted pattern to have all tests and test configuration files on a root `tests` folder instead of next to each sub-section, so I vote to move all to `data_pipeline/tests`

**Comment by lucasmbrown-usds** [comment]
Yeah I have no strong preference. At my last org we kept tests next to code which made it easy to find the tests when you needed them, but (as usual) all I really care about is consistency and parallelism :), so moving them all to a separate root tests folder is fine for me. 

I assume the separate tests folder would mirror the directory structure of the application code?

**Comment by saran-ahluwalia** [comment]
@esfoobar-usds Here are some suggestions from a couple references:

1. Python Testing Reference [structure](https://dssg.github.io/hitchhikers-guide/curriculum/programming_best_practices/test-test-test/python_testing/) 
2. Pytest Suggestions and conventions used for [modules](https://docs.pytest.org/en/6.2.x/goodpractices.html)

**Comment by sverchdotgov** [comment]
This is what I think the steps are here:

- Review the [current testing documentation](https://github.com/usds/justice40-tool/tree/main/data/data-pipeline#testing) and check how closely the reality of how we test matches that documentation.
- Create a PR with any docs fixes and with documentation of what the directory structure should be, so that we can get to consensus on what that should be.
- Once we agree on the PR, merge in and then update the test structure to match what's documented.
